runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -299,6 +299,16 @@ function M.on_diagnostic(_, result, ctx, config)
   local fname = vim.uri_to_fname(uri)
 
   if result == nil then
+    local client_name = vim.lsp.get_client_by_id(client_id).name
+    vim.notify_once(
+      string.format(
+        "LSP: client '%s' (id %s) returned nil to textDocument/diagnostic.\nThis is a bug in the LSP backend, and should be fixed within '%s'.",
```

Comment:
I would use "server" instead of "backend", as that's a more common term in LSP lingo. 
```suggestion
        "LSP: client '%s' (id %s) returned nil to textDocument/diagnostic.\nThis is a bug in the LSP server, and should be fixed within '%s'.",
```